packit/utils/koji_helper.py

```diff
@@ -14,18 +14,54 @@
 logger = logging.getLogger(__name__)
 
 
-class KojiHelper:
-    """
-    Class for querying Koji.
+class SessionWrapper:
```

Comment:
Wouldn't this be better done as a context manager? I have a feeling it could also simplify handling of the exceptions

Reply:
But then I would have to change every API call, right? I was trying to avoid that, but maybe it's woth it. What do you think?

Reply:
I think it could simplify also the code, the  `__getattr__`  override shouldn't be needed with a context manager, right?

Reply:
That's true, the `SessionWrapper` is just a way to avoid changing majority of the code (every call of a method of `koji.ClientSession`).

Reply:
It feels like more correct way, but not sure :shrug:

Reply:
Actually, @mfocko how do you imagine a context manager would work here?

Reply:
Did you mean opening a new session for each API call?

Reply:
> Did you mean opening a new session for each API call?

that would resolve the issue for sure…

I'm not sure if I prefer the readability of this, or copy-paste of

```py
with KojiSessionWrapper() as koji:
  …
```

Reply:
> that would resolve the issue for sure…

That would be the cleanest solution, but we could hit some undocumented (?) rate limits.

> I'm not sure if I prefer the readability of this, or copy-paste of
> 
> ```python
> with KojiSessionWrapper() as koji:
>   …
> ```

So still using a new session for every API call? Looking at the Koji code, unauthenticated session seems to be just a wrapper over `requests.Session`, authenticated sessions are tracked server-side and have expiration etc., but I'm not able to tell if there are any limits on the number of sessions per user/client.

Reply:
I was also thinking about separating authenticated (only 4, 2 of those really used currently) and unanthenticated calls, keep only authenticated session open until it expires and use throw-away sessions for majority of the calls. Does it make any sense?

Reply:
I think we could also do that… it's kinda shame that the Koji doesn't separate this itself…

Reply:
I think I'll keep the code as it is for now, we can change it if we experience issues.